### ISSUE TITLE
Pipeline の _last_letterbox_params インスタンス状態を request-scoped に分離

### ISSUE THREAD

**Issue by kurorosu** [kurorosu/pochidetection#569]
## Summary

`RTDetrPipeline` / `SsdPipeline` の `_last_letterbox_params` (インスタンス状態) を撤去し, preprocess → postprocess の受け渡しを明示的に行う. 現状は WebAPI のマルチリクエスト環境で race condition の火種になっており, train/inference どちらの経路でも同一 pipeline インスタンスを並行呼出しする際に bbox 座標のズレが生じる可能性がある.

## Background

[#567](https://github.com/kurorosu/pochidetection/pull/567) で推論 letterbox を導入した際に, preprocess で計算した `LetterboxParams` を postprocess へ受け渡すため `self._last_letterbox_params` というインスタンス属性を導入した. 同時に既存の `_gpu_input_buffer` も同種の pipeline-scoped state.

- 単スレッド CLI / 単スレッド ストリーム推論では問題ない
- FastAPI の `pochi serve` は sync endpoint + threadpool で複数リクエストを並行処理する. 同一 pipeline インスタンスを共有するため, リクエスト A の preprocess が params を書き込んでから postprocess 実行までの間に, リクエスト B が preprocess を実行すると params が上書きされる
- `_gpu_input_buffer` も同様だが, shape が同じ場合は問題が顕在化しにくい (偶然動いているだけ)

## Current Implementation

`pochidetection/pipelines/rtdetr_pipeline.py::preprocess`:

```python
# letterbox 用の幾何パラメータを事前計算 (元画像サイズ必要).
if self._letterbox and self._target_hw is not None:
    ...
    self._last_letterbox_params = compute_letterbox_params((src_h, src_w), self._target_hw)
else:
    self._last_letterbox_params = None
```

`pochidetection/pipelines/rtdetr_pipeline.py::postprocess`:

```python
params = self._last_letterbox_params
if params is not None and self._target_hw is not None:
    ...
```

同じパターンが [ssd_pipeline.py](pochidetection/pipelines/ssd_pipeline.py) にも存在.

### 案 A: preprocess の戻り値に params を含める

preprocess の戻り値を `(inputs_dict, params)` のタプル (または `PreprocessResult` dataclass) に変更し, `run()` で params を postprocess へ明示的に渡す.

```python
def preprocess(self, image: ImageInput) -> tuple[dict[str, torch.Tensor], LetterboxParams | None]:
    ...
    return inputs, params

def postprocess(
    self, pred_logits, pred_boxes, image_size, threshold=None, letterbox_params=None
) -> list[Detection]:
    ...
```

`run()` が coordination point なので params の lifetime が明確になる.

### 案 B: `PipelineRunState` (request-scoped dataclass) を run() 内でローカル保持

`run()` 内でローカル state を持ち, preprocess / postprocess にメソッド引数として渡す. pipeline instance state を完全排除.

### 案 C: 現状維持 + ドキュメント警告

インスタンス pipeline を複数スレッドから呼ばないよう runtime assertion or doc で明記.

**推奨**: 案 A. signature 変更は局所的で, 影響範囲は `run()` と test の 2-3 箇所. 既存の `(inputs, orig_w, orig_h)` 形式 (SSD pipeline) とも整合性が取れる.

### GPU buffer の扱い

`_gpu_input_buffer` はバッファ再利用による allocation コスト削減が目的で, 設計意図上 pipeline-scoped が正しい. マルチリクエスト時は**別 pipeline インスタンスを使う**運用で回避するのが妥当. ただしドキュメントに「同一 pipeline を複数 thread から呼ばない」と明記する.

## Acceptance Criteria

- `RTDetrPipeline` / `SsdPipeline` から `_last_letterbox_params` インスタンス属性が削除される
- `preprocess` 戻り値に params を含める (または同等の request-scoped 受け渡し)
- `run` 内で preprocess → postprocess に params が明示的に渡される
- 既存のテスト ([test_rtdetr_pipeline.py](tests/test_pipelines/test_rtdetr_pipeline.py), [test_ssdlite_pipeline.py](tests/test_pipelines/test_ssdlite_pipeline.py)) が pass
- マルチ thread 呼出でも params が混線しないことを classical test で検証 (2 並行 thread で異なる shape の画像を渡し, bbox が各画像の座標系に正しく戻るか確認)
- `_gpu_input_buffer` のスレッド非安全性は `RTDetrPipeline` / `SsdPipeline` の docstring に明記される

## Labels

`refactor`

## Branch

`refactor/pipeline-request-scoped-state`
